ee/audit/run_ee.go

```diff
@@ -107,37 +107,161 @@ func run() error {
 		glog.Info("audit file is empty")
 		return nil
 	}
-	var iterator int64 = 0
 
-	iv := make([]byte, aes.BlockSize)
-	x.Check2(file.ReadAt(iv, iterator))
-	iterator = iterator + aes.BlockSize
+	// decrypt header in audit log to verify encryption key
+	// [16]byte IV + [4]byte len(x.VerificationText) + [11]byte x.VerificationText
+	decryptHeader := func() ([]byte, int64, error) {
+		var iterator int64 = 0
+		iv := make([]byte, aes.BlockSize)
+		_, err := file.ReadAt(iv, iterator) // get first iv
+		if err != nil {
+			return nil, 0, err
+		}
+		iterator = iterator + aes.BlockSize + 4 // length of verification text encoded in uint32
 
-	t := make([]byte, len(x.VerificationText))
-	x.Check2(file.ReadAt(t, iterator))
-	iterator = iterator + int64(len(x.VerificationText))
+		ct := make([]byte, len(x.VerificationText))
+		_, err = file.ReadAt(ct, iterator)
+		if err != nil {
+			return nil, 0, err
+		}
+		iterator = iterator + int64(len(x.VerificationText))
 
-	stream := cipher.NewCTR(block, iv)
-	stream.XORKeyStream(t, t)
-	if string(t) != x.VerificationText {
-		return errors.New("invalid encryption key provided. Please check your encryption key")
+		text := make([]byte, len(x.VerificationText))
+		stream := cipher.NewCTR(block, iv)
+		stream.XORKeyStream(text, ct)
+		if string(text) != x.VerificationText {
+			return nil, 0, errors.New("invalid encryption key provided. Please check your encryption key")
+		}
+		return iv, iterator, nil
 	}
 
-	for {
-		// if its the end of data. finish decrypting
-		if iterator >= stat.Size() {
-			break
+	// [12]byte baseIV + [4]byte len(x.VerificationTextDeprecated) + [11]byte x.VerificationTextDeprecated
+	decryptHeaderDeprecated := func() ([]byte, int64, error) {
+		var iterator int64 = 0
+
+		iv := make([]byte, aes.BlockSize)
```

Comment:
The length of the blocksize is 16 in both cases.  Previously we were using a baseIV (12 bytes) appended with the length of the cipher text (4 bytes, uint32) as the IV.  Now we randomly generate a 16 byte IV each time, followed by the length of the cipher text (4 bytes, uint32).  This is the point of the PR, because what we were doing before (baseIV + len(ciphertext)) introduces the possibility of collisions.